### DISCUSSION TITLE
Off Grid - On Battery: Watching a Power Blip on the Dashboard

### DISCUSSION THREAD

**Discussion by jasonacox** [jasonacox/Powerwall-Dashboard#481] [General]
This past week, we had a brief grid ~2 minute outage. Nobody in the house noticed and presumably since the duration was so small, the Tesla App didn't even send out a notification. However, the Dashboard did see it. 

<img width="316" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/4e1266f5-652a-45d6-9f6c-521e065fa8af">

With the new enhanced metrics from TEDAPI (currently in beta) I also got to see the alerts that fire during such an event:

<img width="862" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/67b8f622-6b13-422e-aff1-f6ebdbc45cf8">

Because the grid event was so short, the dashboard averaging and downsampling actually only registers an average dip of grid and Powerwall supply. It may be good to see if there is a way to better capture these short blips.

<img width="565" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/c46741b3-d1da-4266-bee9-781624aeda4a">

However, the grid voltage drop to zero does show up:

<img width="820" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/6e3bd5e9-2d29-4c7e-8536-83405dc08507">

The whole house battery backup is nice. It was fun to see this event with the extended data again.

**Comment by hulkster** [comment]
That's pretty cool.

I wonder if you got "lucky" with the sampling to capture the voltage drop?

LOL that thanks to your battery backup, unless you had PowerWall running (and were pretty attentive), you would not even noticed.

Hopefully your utility company is paying attention and if these events repeat/persist, try to figure out why and fix.

While I don't have a battery backup, a while ago I spent the big bucks on a nice Sine-Wave UPS for my computer system and have it where I can see/hear it ... kinda surprised that a few times/month, it will do a short "switchover" due to drop in voltage ... but NOT enough to cause lights to blip, etc.

**Reply by jasonacox** [reply]
Ha! True!  And yes, I have noticed how "dirty" grid voltage can be.  It generally stays 120-122V but can go outside that range:

<img width="796" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/b9702ce8-7465-4272-8727-2bcc44416ccd">
